src/functor.jl

```diff
@@ -44,10 +44,29 @@ Equivalent to `functor(x)[1]`.
 """
 children(x) = functor(x)[1]
 
+function functor_tuple(f, x::Tuple, dx::Tuple)
```

Comment:
I don't think `x` is a tuple
```suggestion
function functor_tuple(f, x, dx::Tuple)
```